src/useDCPWorker.tsx

```diff
@@ -355,249 +295,277 @@ function getWorkerOptionsKey() {
+ * @param config.identity        Value to set the identity of the worker. Must be of type dcp.wallet.Address. Optional.
+ *                               If not provided, an identity will be generated.
  * @param config.useLocalStorage Boolean flag to enable the use of localStorage to save workerOptions.
- *                               This will override any workerOptions passed in the config.
+ *                               This will override `paymentAddress` and `maxWorkingSandboxes` properties defined in
+ *                               `config.workerOptions`. Optional, set to `true` by default.
  * @param config.workerOptions   WorkerOptions to configure the worker.
- * @returns `{workerState, workerStatistics, setWorkerOptions, startWorker, stopWorker, toggleWorker, workerOptionsState, sandboxes}`
- *          workerState and workerStatisitics provide worker status and data information. The worker can be controlled by startWorker,
- *          stopWorker and togglerWorker. workerOptionsState is a readonly object that describes how the worker is currently
- *          configured. To mutate workerOptions, use setWorkerOptions.
+ * @returns `{worker, workerOptions, workerState, workerStatistics}`
  */
-const useDCPWorker = ({
-  identity = null,
-  useLocalStorage = true,
-  workerOptions: userWorkerOptions,
-}: IUseDCPWorkerParams) => {
+const useDCPWorker = (
+  {
+    identity = null,
+    useLocalStorage = true,
+    workerOptions: userWorkerOptions,
+  }: IUseDCPWorkerParams
+) => {
   const {
     worker,
     setWorker,
-    workerOptionsState,
-    setWorkerOptionsState,
     workerState,
     workerStatistics,
     dispatchWorkerState,
     dispatchWorkerStats,
   } = useContext(WorkerContext);
 
-  function startWorker() {
-    if (workerState.isLoaded && worker !== null) {
-      dispatchWorkerState({ type: 'WILL_WORK_TRUE' });
-      worker.start().catch((error: any) => {
-        console.error(
-          `use-dcp-worker: starting the worker threw an unexpected error:`,
-          error,
-        );
-        return error;
-      });
-    } else console.warn('use-dcp-worker: worker is not loaded.');
-  }
-
-  function stopWorker() {
-    dispatchWorkerState({ type: 'WILL_WORK_FALSE' });
-    if (workerState.isLoaded && worker !== null) {
-      worker
-        .stop(workerOptions.shouldStopWorkingImmediately ?? false)
-        .catch((error: any) => {
-          console.error(
-            `use-dcp-worker: stopping the worker threw an unexpected error:`,
-            error,
-          );
-          return error;
-        });
-    } else {
-      console.error(
-        "use-dcp-worker: failed to stop worker, worker isn't loaded or is null.",
-      );
+  /**
+   *  This method ensures that the paymentAddress prop in workerOptions is of valid type.
+   *  @returns `true` if valid, `false` otherwise
+   */
+  const ensurePaymentAddressType = useCallback(() => {
```

Comment:
**issue:** If I don't set the payment address, there's currently no way at runtime to handle the situation.

Why don't we throw an error on the invalid option being passed in? Or at the very least, return the error somehow so that consumers can decide what to do in this situation.

We throw an error when `window.dcp` isn't available. The lack of an address should be treated in a similar manner imo.

Reply:
Currently the error is logged to the console. Currently if an invalid or missing paymentAddress is provided then the hook wont pass the constructWorkerOptions stage. However, if the the value for paymentAddress gets defined/corrected during a re-render, then the hook will re-attempt constructing. In the case where using `useState` for the value of paymentAddress in the hook params, the state may not be defined with a valid value by default. In that case, if I was to throw, then the app would not be in a recoverable state. Similarly, `ensurePaymentAddressType` is called on changes to the worker.workerOptions.paymentAddress - I don't want to throw in those cases either.

Reply:
Ah, okay. I didn't grok that from reading the code. 

**suggestion:** Document the hook's behaviour when an invalid address is passed. i.e., describe what a consumer would need to do to handle that situation through the use of `isLoaded`. Because it's not clear to me how one should handle an erroneous condition from just reading the code/docs.

I'd also like to fully address the first half of https://github.com/Distributive-Network/use-dcp-worker/pull/2#discussion_r1245730062 as well, since it's still unclear from skimming the docs which options are mandatory or not. I know it's implied from the example, but making it more obvious in the reference docs would be great!

Reply:
I added a note under the paymentAddress param desc